src/components/application_manager/test/commands/mobile/change_registration_test.cc

```diff
@@ -419,18 +456,65 @@ TEST_F(ChangeRegistrationRequestTest, Run_IsLanguageSupportedByUITrue_SUCCESS) {
   command_->Run();
 }
 
+TEST_F(ChangeRegistrationRequestTest,
+       Run_IsLanguageSupportedByVRLanguagesEmpty_UNSUCCESS) {
+  CreateDefaultMessage(message_);
+  const utils::custom_string::CustomString name(kAppName2);
+  application_set_.insert(app_);
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(app_));
+  EXPECT_CALL(mock_app_manager_, applications())
+      .WillOnce(Return(applications_));
+  EXPECT_CALL(*app_, name()).WillOnce(ReturnRef(name));
+
+  MessageSharedPtr vr_synonyms_message(CreateMessage());
+  (*vr_synonyms_message)[am::strings::msg_params][am::strings::vr_synonyms][0] =
+      kVrSynonyms2;
+  smart_objects::SmartObject& vr_synonyms =
+      (*vr_synonyms_message)[am::strings::msg_params][am::strings::vr_synonyms];
+  EXPECT_CALL(*app_, vr_synonyms())
+      .WillOnce(Return(static_cast<smart_objects::SmartObject*>(&vr_synonyms)));
```

Comment:
@okozlovlux, `static_cast` here is redundant and incorrect(you cast `SharedPtr<SmartObject>*` to  `SmartObject*`), please, rather use `vr_synonyms.get()`
Also, check everywhere, I saw this kind of issue in other places too